packages/core-p2p/src/peer-communicator.ts

```diff
@@ -289,12 +289,16 @@ export class PeerCommunicator implements Contracts.P2P.PeerCommunicator {
 
     private async throttle(peer: Contracts.P2P.Peer, event: string): Promise<void> {
         const msBeforeReCheck = 1000;
-        while (await this.outgoingRateLimiter.hasExceededRateLimit(peer.ip, event)) {
+        while (await this.outgoingRateLimiter.hasExceededRateLimitNoConsume(peer.ip, event)) {
             this.logger.debug(
                 `Throttling outgoing requests to ${peer.ip}/${event} to avoid triggering their rate limit`,
             );
             await delay(msBeforeReCheck);
         }
+        try {
+            await this.outgoingRateLimiter.hasExceededRateLimit(peer.ip, event);
```

Comment:
`this.outgoingRateLimiter.hasExceededRateLimit` [doesn't throw]. It returns either `true` or `false`.

[doesn't throw]: https://github.com/ArkEcosystem/core/blob/develop/packages/core-p2p/src/rate-limiter.ts#L41-L57